src/middlewares/autorization/auth-util.ts

```diff
@@ -0,0 +1,59 @@
+import { NextFunction, Request, Response } from 'express';
+import {authStrategy, securedApi} from '../../config.js';
+import egoAuth from './ego-auth-handler.js';
+import keycloakAuth from './keycloak-auth-handler.js';
+import { ForbiddenError, UnauthorizedError } from '../error-handler';
+
+export interface AuthorizationStrategy {
+	authHandler(req: Request, res: Response, next: NextFunction): Promise<void>;
+}
+
+function getAuthStrategy() {
+	if (authStrategy === 'EGO') {
+		return egoAuth;
+	} else if (authStrategy === 'KEYCLOAK') {
+		return keycloakAuth;
+	}
+}
+
+/// ego-KC switch
+export function authorize(action: string): MethodDecorator {
+	console.log('authorize called');
+	return (target: any, propertyKey: string | symbol, descriptor: PropertyDescriptor) => {
+		const origFunction = descriptor.value;
+		descriptor.value = async function (...args: any[]) {
+			const request = arguments[0] as Request;
+			const response = arguments[1] as Response;
+			const next = arguments[2] as NextFunction;
+			try {
+				if(securedApi.length == 0 || securedApi.includes(action)) await getAuthStrategy()?.authHandler(request, response, next);
+				origFunction.apply(this, args);
+			} catch (err) {
+				next(err);
+			}
+		};
+	};
+}
+
+export function extractHeaderToken(req: Request) {
+	const { authorization: authorizationHeader } = req.headers;
+	const { authorization: authorizationBody } = req.body || {};
+	const authorization = authorizationHeader || authorizationBody;
```

Comment:
Why are we taking auth info from the body?

Reply:
we don't need that. I added it when I started and didn't clean it up later.  Thanks.